src/test/javascript/spec/component/shared/submission-policy-update.component.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { LockRepositoryPolicy, SubmissionPenaltyPolicy, SubmissionPolicyType } from 'app/entities/submission-policy.model';
+import { SubmissionPolicyUpdateComponent } from 'app/exercises/shared/submission-policy/submission-policy-update.component';
+import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
+import { MockComponent, MockDirective, MockPipe } from 'ng-mocks';
+import { TranslateDirective } from 'app/shared/language/translate.directive';
+import { HelpIconComponent } from 'app/shared/components/help-icon.component';
+
+describe('Submission Policy Update Form Component', () => {
+    const lockRepositoryPolicy = { type: SubmissionPolicyType.LOCK_REPOSITORY, submissionLimit: 5 } as LockRepositoryPolicy;
+    const submissionPenaltyPolicy = { type: SubmissionPolicyType.SUBMISSION_PENALTY, submissionLimit: 5, exceedingPenalty: 50.4 } as SubmissionPenaltyPolicy;
+    const brokenPenaltyPolicy = { type: SubmissionPolicyType.SUBMISSION_PENALTY } as SubmissionPenaltyPolicy;
+
+    let fixture: ComponentFixture<SubmissionPolicyUpdateComponent>;
+    let component: SubmissionPolicyUpdateComponent;
+    let expectedProgrammingExercise: ProgrammingExercise;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [FormsModule, ReactiveFormsModule],
+            declarations: [SubmissionPolicyUpdateComponent, MockPipe(ArtemisTranslatePipe), MockDirective(TranslateDirective), MockComponent(HelpIconComponent)],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(SubmissionPolicyUpdateComponent);
+        component = fixture.componentInstance;
+
+        expectedProgrammingExercise = new ProgrammingExercise(undefined, undefined);
+        expectedProgrammingExercise.submissionPolicy = lockRepositoryPolicy;
+        component.programmingExercise = expectedProgrammingExercise;
+    });
+
+    it('Should set policy object on exercise', fakeAsync(() => {
+        component.programmingExercise.submissionPolicy = undefined;
+        component.ngOnInit();
+        fixture.detectChanges();
+        tick();
+
+        expect(expectedProgrammingExercise.submissionPolicy).toBeUndefined();
+
+        const submissionPolicyTypeField = fixture.nativeElement.querySelector('#field_submissionPolicy');
+        // We revert the enum values, since 'none' is the default type for the type picker. Therefore we
+        // need to set the picker to anything other than 'none' to test switching back to 'none' appropriately
+        for (const type of Object.values(SubmissionPolicyType).reverse()) {
+            submissionPolicyTypeField.value = type;
+            submissionPolicyTypeField.dispatchEvent(new Event('change'));
+            fixture.detectChanges();
+            tick();
+
+            expect(expectedProgrammingExercise.submissionPolicy?.type).toBe(type);
+            expect(expectedProgrammingExercise.submissionPolicy?.id).toBeUndefined();
```

Comment:
```suggestion
            expect(expectedProgrammingExercise.submissionPolicy?.id).toBe(undefined);
```
according to our guidelines

Reply:
Fixed